core/trino-main/src/main/java/io/trino/connector/system/RuleStatsSystemTable.java

```diff
@@ -50,10 +48,10 @@
 {
     private static final SchemaTableName TABLE_NAME = new SchemaTableName("runtime", "optimizer_rule_stats");
     private final ConnectorTableMetadata ruleStatsTable;
-    private final Optional<RuleStatsRecorder> ruleStatsRecorder;
+    private final RuleStatsRecorder ruleStatsRecorder;
 
     @Inject
-    public RuleStatsSystemTable(Optional<RuleStatsRecorder> ruleStatsRecorder, Metadata metadata)
```

Comment:
Q: was this coming up empty even after singleton binding because another instance already existed?

Reply:
No, I didn't try that, but I see it works. So let me add that and revert this part of the change.